File: RQ2_Kotlin/src/main/java/ch/uzh/ifi/seal/smr/reconfigure/statistics/ci/OpenCSVWriter.java
package ch.uzh.ifi.seal.smr.reconfigure.statistics.ci;

import com.opencsv.CSVWriter;
import com.opencsv.bean.StatefulBeanToCsv;
import com.opencsv.bean.StatefulBeanToCsvBuilder;
import com.opencsv.exceptions.CsvDataTypeMismatchException;
import com.opencsv.exceptions.CsvRequiredFieldEmptyException;

import java.io.File;
import java.io.IOException;
import java.io.Writer;
import java.nio.file.Files;
import java.util.List;

public class OpenCSVWriter {
    public static void write(File outputFile, List<Input> input){
        try {
            if (!outputFile.exists()) {
                outputFile.createNewFile();
            }

            Writer writer = Files.newBufferedWriter(outputFile.toPath());

            StatefulBeanToCsvBuilder<Input> builder = new StatefulBeanToCsvBuilder<Input>(writer)
                    .withQuotechar(CSVWriter.NO_QUOTE_CHARACTER)
                    .withSeparator(';')
                    .withMappingStrategy(new CustomMappingStrategy(Input.class));

            StatefulBeanToCsv<Input> beanToCsv = builder.build();
            beanToCsv.write(input);

            writer.close();
        }catch (IOException | CsvRequiredFieldEmptyException | CsvDataTypeMismatchException e){
            e.printStackTrace();
        }
    }
}

File: State_of_Practise_Code/src/main/kotlin/ch/uzh/ifi/seal/smr/soa/evaluation/OpenCSVWriter.kt
package ch.uzh.ifi.seal.smr.soa.evaluation

import com.opencsv.CSVWriter
import com.opencsv.bean.StatefulBeanToCsvBuilder

import java.nio.file.Files
import java.nio.file.Path

object OpenCSVWriter {
    fun <T> write(outputFile: Path, input: Iterable<T>) {
        if (!outputFile.toFile().exists()) {
            outputFile.toFile().createNewFile()
        }

        Files.newBufferedWriter(outputFile).use { writer ->
            val beanToCsv = StatefulBeanToCsvBuilder<T>(writer)
                    .withQuotechar(CSVWriter.NO_QUOTE_CHARACTER)
                    .build()

            beanToCsv.write(input.toList())
        }
    }
}